odilo.py

```diff
@@ -0,0 +1,581 @@
+            identifier = record_id.identifier
+
+        url = self.RECORD_METADATA_ENDPOINT.replace('{recordId}', identifier)
+
+        response = self.get(url)
+
+        if response and response.status == 200 and response.content:
+            return response.content
+        else:
+            self.log.warn('Cannot retrieve metadata for record: ' + record_id + ' response http ' + response.status)
+            if response.content:
+                self.log.warn(response.content)
+            return None
+
+    def get_availability(self, record_id):
+        url = self.RECORD_AVAILABILITY_ENDPOINT.replace('{recordId}', record_id)
+        status_code, headers, content = self.get(url)
+        content = json.loads(content)
+
+        if status_code == 200 and len(content) > 0:
+            return content
+        else:
+            self.log.warn('Cannot retrieve availability for record: ' + record_id + ' response http ' + status_code)
+            if content:
+                self.log.warn(content)
+            return None
+
+    @staticmethod
+    def _do_get(url, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        if 'allow_redirects' not in kwargs:
+            kwargs['allow_redirects'] = True
+
+        response = HTTP.get_with_timeout(url, headers=headers, **kwargs)
+        return response.status_code, response.headers, response.content
+
+    @staticmethod
+    def _do_post(url, payload, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        return HTTP.post_with_timeout(url, payload, headers=headers, **kwargs)
+
+
+class MockOdiloAPI(OdiloAPI):
+    @classmethod
+    def mock_collection(cls, _db):
+        library = DatabaseTest.make_default_library(_db)
+        collection, ignore = get_one_or_create(
+            _db, Collection,
+            name="Test Odilo Collection",
+            create_method_kwargs=dict(
+                external_account_id=u'library_id_123',
+            )
+        )
+        integration = collection.create_external_integration(
+            protocol=ExternalIntegration.ODILO
+        )
+        integration.password = u'abcdef123hijklm'
+        library.collections.append(collection)
+        return collection
+
+
+class OdiloRepresentationExtractor(object):
+    """Extract useful information from Odilo's JSON representations."""
+
+    log = logging.getLogger("OdiloRepresentationExtractor")
+
+    format_data_for_odilo_format = {
+        "ACSM": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM
+        ),
+        "PDF": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "EPUB": (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
```

Comment:
I don't believe a PDF or EPUB file can be delivered through streaming. If this is a PDF or EPUB file that's been unpacked, rendered to HTML+images and delivered through a web application, then the media type here should be HTML_MEDIA_TYPE. (It will be delivered to the client as `text/html;profile=http://librarysimplified.org/terms/profiles/streaming-media` to distinguish it from a normal "book in a single web page".

If this is a PDF or EPUB that is downloaded in full onto the client and stored there, then the media type is correct but the DRM type is NO_DRM.

Reply:
Ok, It's the first case you suppose. I have already correct the media type for both PDF and EPUB files.